src/ItemTreeTraversal/MysqlItemTreeTraversal.php

```diff
@@ -92,21 +92,22 @@ public function hasChild(string $parentName, string $childName): bool
             ->select([new Expression('1 AS item_child_exists')])
             ->andWhere(['item.name' => $childName]);
         /** @psalm-var array<0, 1>|false $result */
-        $result = $this->getChildrenRowsStatement($parentName, baseOuterQuery: $baseOuterQuery)->queryScalar();
+        $result = $this->getChildrenRowsCommand($parentName, baseOuterQuery: $baseOuterQuery)->queryScalar();
 
         return $result !== false;
     }
 
-    private function getChildrenRowsStatement(string $name, QueryInterface $baseOuterQuery): CommandInterface
+    private function getChildrenRowsCommand(string $name, QueryInterface $baseOuterQuery): CommandInterface
     {
         $fromSql = "SELECT DISTINCT child
         FROM (SELECT * FROM $this->childrenTableName ORDER by parent) item_child_sorted,
         (SELECT @pv := :name) init
         WHERE find_in_set(parent, @pv) AND length(@pv := concat(@pv, ',', child))";
         $outerQuery = $baseOuterQuery
             ->from(new Expression("($fromSql) s"))
-            ->leftJoin($this->tableName . ' AS item', ['item.name' => new Expression('s.child')]);
+            ->leftJoin($this->tableName . ' AS item', ['item.name' => new Expression('s.child')])
+            ->addParams([':name' => $name]);
 
-        return $this->database->createCommand($outerQuery->createCommand()->getSql(), [':name' => $name]);
+        return $this->database->createCommand($outerQuery->createCommand()->getRawSql());
```

Comment:
Better return back
https://github.com/yiisoft/db/blob/master/src/Command/CommandInterface.php#L498-L509

Reply:
Yes, I read this before. But it's tested and works so far. What's the alternative of getting SQL with parameters' actual values then? Otherwise the parameters need to be passed twice - in the first one with subquery it won't have any effect, which hurts readability in my opinion.

Reply:
Suggested alternative below.